examples/dynamic_mat_transpose.cpp

```diff
@@ -0,0 +1,439 @@
+ *    - Basic usage of 'RAJA::launch' abstractions for nested loops
+ *    - Hierachial parallism
+ *    - Dynamic shared memory
+ *
+ * If CUDA is enabled, CUDA unified memory is used.
+ */
+
+//
+// Define dimensionality of matrices
+//
+const int DIM = 2;
+
+//
+// Function for checking results
+//
+template <typename T>
+void checkResult(RAJA::View<T, RAJA::Layout<DIM>> Atview, int N_r, int N_c);
+
+//
+// Function for printing results
+//
+template <typename T>
+void printResult(RAJA::View<T, RAJA::Layout<DIM>> Atview, int N_r, int N_c);
+
+using launch_policy = RAJA::LaunchPolicy<
+#if defined(RAJA_ENABLE_OPENMP)
+    RAJA::omp_launch_t
+#else
+    RAJA::seq_launch_t
+#endif
+#if defined(RAJA_ENABLE_CUDA)
+    ,
+    RAJA::cuda_launch_t<false>
+#endif
+#if defined(RAJA_ENABLE_HIP)
+    ,
+    RAJA::hip_launch_t<false>
+#endif
+#if defined(RAJA_ENABLE_SYCL)
+    ,
+    RAJA::sycl_launch_t<false>
+#endif
+    >;
+
+/*
+ * Define team policies.
+ * Up to 3 dimension are supported: x,y,z
+ */
+using outer0 = RAJA::LoopPolicy<
+                                       RAJA::loop_exec
+#if defined(RAJA_ENABLE_CUDA)
+                                       ,
+                                       RAJA::cuda_block_x_direct
+#endif
+#if defined(RAJA_ENABLE_HIP)
+                                       ,
+                                       RAJA::hip_block_x_direct
+#endif
+#if defined(RAJA_ENABLE_SYCL)
+                                       ,
+                                       RAJA::sycl_group_0_direct
+#endif
+                                       >;
+
+using outer1 = RAJA::LoopPolicy<
+#if defined(RAJA_ENABLE_OPENMP)
+                                       RAJA::omp_for_exec
+#else
+                                       RAJA::loop_exec
+#endif
+#if defined(RAJA_ENABLE_CUDA)
+                                       ,
+                                       RAJA::cuda_block_y_direct
+#endif
+#if defined(RAJA_ENABLE_HIP)
+                                       ,
+                                       RAJA::hip_block_y_direct
+#endif
+#if defined(RAJA_ENABLE_SYCL)
+                                       ,
+                                       RAJA::sycl_group_1_direct
+#endif
+                                       >;
+/*
+ * Define thread policies.
+ * Up to 3 dimension are supported: x,y,z
+ */
+using inner0 = RAJA::LoopPolicy<
+#if defined(RAJA_ENABLE_OPENMP)
+                                         RAJA::omp_for_exec
+#else
+                                         RAJA::loop_exec
+#endif
+
+#if defined(RAJA_ENABLE_CUDA)
+                                         ,
+                                         RAJA::cuda_thread_x_direct
+#endif
+#if defined(RAJA_ENABLE_HIP)
+                                         ,
+                                         RAJA::hip_thread_x_direct
+#endif
+#if defined(RAJA_ENABLE_SYCL)
+                                        ,
+                                         RAJA::sycl_local_0_direct
+#endif
```

Comment:
For the next release we should create the RAJA alias' for the gpu policies

Reply:
What do you mean here? The generic ones or something else?

Reply:
I figure most codes will have this switch-yard, so we can help codes a bit by doing it for them.